misc/swagger-ballerina/modules/swagger-to-ballerina-generator/src/main/java/org/ballerinalang/swagger/CodeGenerator.java

```diff
@@ -0,0 +1,146 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ballerinalang.swagger;
+
+import com.github.jknack.handlebars.Context;
+import com.github.jknack.handlebars.Handlebars;
+import com.github.jknack.handlebars.Template;
+import com.github.jknack.handlebars.context.FieldValueResolver;
+import com.github.jknack.handlebars.helper.StringHelpers;
+import com.github.jknack.handlebars.io.ClassPathTemplateLoader;
+import com.github.jknack.handlebars.io.FileTemplateLoader;
+import io.swagger.oas.models.OpenAPI;
+import io.swagger.parser.v3.OpenAPIV3Parser;
+import org.ballerinalang.swagger.utils.GeneratorConstants;
+import org.ballerinalang.swagger.utils.GeneratorConstants.GenType;
+import org.ballerinalang.swagger.utils.OpenApiWrapper;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+/**
+ * <p>This class generates Ballerina Services/Connectors for a provided OAS definition.</p>
+ */
+public class CodeGenerator {
+    private String apiPackage;
+
+    /**
+     * Generates ballerina source for provided Open API Definition in <code>definitionPath</code>
+     * <p>Method can be user for generating Ballerina service skeletons, mock services and connectors</p>
+     *
+     * @param type           Output type. Following types are supported
+     *                       <ul>
+     *                       <li>skeleton</li>
+     *                       <li>mock</li>
+     *                       <li>connector</li>
+     *                       </ul>
+     * @param definitionPath Input Open Api Definition file path
+     * @param outPath        Destination file path to save generated source files. If not provided
+     *                       <code>destinationPath</code> will be used as the default destination path
+     * @throws IOException when file operations fail
+     */
+    public void generate(GenType type, String definitionPath, String outPath) throws IOException {
+
+        OpenAPI api = new OpenAPIV3Parser().read(definitionPath);
+        OpenApiWrapper context = new OpenApiWrapper().buildFromOpenAPI(api).apiPackage(apiPackage);
+
+        // Write output to the input definition location if no destination directory is provided
+        if (outPath == null || "".equals(outPath)) {
+            String fileName = api.getInfo().getTitle().replaceAll(" ", "") + ".bal";
+            outPath = definitionPath.substring(0, definitionPath.lastIndexOf(File.separator) + 1);
+            outPath += fileName;
+        }
+        switch (type) {
+        case SKELETON:
```

Comment:
I think we should indent `case` since it increases readability and it is the format we use.

Reply:
Fixed with 8f4d68b